clusterman/draining/queue.py

```diff
@@ -415,34 +434,44 @@ def host_from_instance_id(
         logger.warning(f"No instance data found for {instance_id}")
         return None
     try:
-        sfr_ids = [tag["Value"] for tag in instance_data[0]["Tags"] if tag["Key"] == "aws:ec2spot:fleet-request-id"]
+        group_ids = [
+            tag["Value"]
+            for tag in instance_data[0]["Tags"]
+            if tag["Key"] in {"aws:ec2spot:fleet-request-id", "aws:ec2:fleet-id", }
+        ]
         scheduler = "mesos"
         for tag in instance_data[0]["Tags"]:
             if tag["Key"] == "KubernetesCluster":
                 scheduler = "kubernetes"
                 break
     except KeyError as e:
-        logger.warning(f"SFR tag key not found: {e}")
-        sfr_ids = []
-    if not sfr_ids:
-        logger.warning(f"No SFR ID found for {instance_id}")
+        logger.warning(f"SFR/SF tag key not found: {e}")
+        group_ids = []
+    if not group_ids:
+        logger.warning(f"No SFR/SF ID found for {instance_id}")
         return None
     try:
         ip = instance_data[0]["PrivateIpAddress"]
     except KeyError:
         logger.warning(f"No primary IP found for {instance_id}")
         return None
+    try:
+        agent_id = instance_data[0]["PrivateDnsName"]
```

Comment:
Yeah, I list EC2s in AWS UI, sort (asc, desc) by Private DNS Name, I saw that all them have it. Is it enough?

Reply:
PrivateDnsName - agent_id is optional now. We will send these nodes (without PrivateDnsName) to termination directly.

Reply:
yea - I was mostly just curious as to when this would not be set, but the AWS docs say `This element remains empty until the instance enters a running state` so this is fine :)